app/controllers/pets_controller.rb

```diff
@@ -20,8 +20,12 @@ def new
   def create
     shelter = Shelter.find(params[:id])
     pet = shelter.pets.create(pet_params)
-    pet.save
-    redirect_to "/shelters/#{shelter.id}/pets"
+    if pet.save
+      redirect_to "/shelters/#{shelter.id}/pets"
+    else
```

Comment:
Adds a flash notice if pets are created without all information, it redirects them back to create page